### ISSUE TITLE
Add `test` value to device.system_tags

### ISSUE THREAD

**Issue by pral2a** [fablabbcn/smartcitizen-api#188]
We're looking for a neat way to filter those device that are only used by tests purposes.

We could use `device.user_tags` but this means a user could easily add the tag to their devices by mistake as any other tag.

We suggest to create a new property inside the `system_tags` called `test` that only will be mutable by users with role "researcher" and "admin" by simply doing a REST operation on `/v0/devices/...`

```json
"system_tags": [
  "offline",
  "outdoor",
  "test"
]
```

After we can exclude them from the platform map by simply performing a front-end filter on `system_tags.test` or by adding filters in `/devices/world_map` in backend

**Comment by viktorsmari** [comment]
This is the current `device.system_tags` method:

https://github.com/fablabbcn/smartcitizen-api/blob/4706778e5a657acfb689ad262c179f972c98a660/app/models/device.rb#L126-L132

It is **not** a variable in the database, but a method which uses logic to see if devices are 'offline' / 'online' / 'new'.

If we want to create our own `system_tags` I think we need to create a new attribute to save in the database, because we probably want more tags in the future.

Idea: Maybe we can add a `boolean` to our current `devices_tags` table called `is_system_tag` (or similar) - where only users with a `role_mask` higher than X can create?

### Thoughts
If User A (researcher) creates a `system_tag` called 'a-test'
Can User B (researcher) edit that tag? Can researcher edit / delete each others tags?

Can users own their own tags and no one else edit them?

**Comment by pral2a** [comment]
Thank you @viktorsmari for such a detailed assessment

We can maybe also consider to simply add another boolean device property like `is_test` or `test_device`

We could even consider this property can be modified by any user owning the device or admin but simply don't expose it on the UI to avoid confusions 

Then we can also add it to `system_tags` just for visibility

**Comment by pral2a** [comment]
Just to recap a bit:

- [ ] Implement `is_test` device boolean flag
- [ ] Add filter to `world_map` to remove devices flagged `is_test` from it
- [ ] Add the flag to existing devices based on some criteria to be confirmed with @oscgonfer ### _i.e. devices owned by user `team`_

**Comment by viktorsmari** [comment]
TLDR: could we use the `is_private` instead?

### Final thoughts:

1. To confirm, this boolean should only hide devices from the World Map?
2. Should it also hide the 'show' page (for other users) like this: https://api.smartcitizen.me/devices/10653 ?
3. Why are we showing `is_private` devices on the World Map? (I don't remember 'why' we need it, I will document the answer in [ADR](https://github.com/fablabbcn/smartcitizen-api/tree/master/docs/adr)).

Currently we have 5 `is_private` devices according to our [metrics](https://api.smartcitizen.me/metrics)

These devices error in the Angular app:
https://smartcitizen.me/kits/10653

![2021-01-04_17-25-30](https://user-images.githubusercontent.com/1689020/103556444-12b61480-4eb2-11eb-8a0b-e475ad93a230.png)

A private device is not very useful on the World Map - why not simply hide it?

### Suggestions
- A: Hide `is_private` on the World Map and use that instead of creating a new boolean?
- B: Create a new boolean as discussed before, but choose a name that reflects the purpose.

Name suggestions:
1. `is_test`  <-- Might be confusing, test can mean many things and it does not say it means it's hidden somewhere.
2. `is_hidden`  <-- Is it hidden everywhere or only in some places?
3. `is_hidden_on_world_map` <-- Too long?

**Comment by pral2a** [comment]
**World Map and private Devices**

We show `is_private` because we didn't want to afford returning a custom `world_map` per user including public and the private devices only visible to him. That way you can still browse the map and find your device, but you only can see the maybe sensitive data once you own it. That feature is only used by 5 devices because the platform admin needs to enable you, it's a paid premium feature. i.e. next February we will deploy 50 devices in Bologna that will use it

**Test Devices**

It's true that if we add support for a custom `world_map` per user as described above we could omit the test devices idea, and I'm not against doing so. However, there is one case you could benefit from an explicit test property. You could use it to remove those devices periodically using a TTL. So, test devices are removed after a month if there isn't new data. Nothing you can't do with private devices, but it will simply take more time.

**Naming**

I don't know up to what point the `is_test` needs to be called hidden. I don't think our API naming need to match the functionality front-end ends up using something for. Maybe the property is also used by our data science framework or by other consumers that use `is_test` in a complete different way. However, it's true `is_test` maybe isn't a great name and other are better, maybe something more linked to `is_internal`...

So, I'm not pushing for one or another just adding some insights...

**Comment by pral2a** [comment]
Great work! @viktorsmari, could you work with @oscgonfer to flag as test existing devices based on some criteria i.e. devices owned by user team

**Comment by oscgonfer** [comment]
Finally a clean IAAC view:

![image](https://user-images.githubusercontent.com/13301321/104952441-c1834600-59c4-11eb-835c-a0435a20c4e3.png)
